ds3-autogen-utils/src/main/java/com/spectralogic/ds3autogen/utils/ResponseTypeConverter.java

```diff
@@ -0,0 +1,257 @@
+        }
+        return builder.build();
+    }
+
+    /**
+     * Updates a Ds3ResponseType so that it will not have a component type. If
+     * there is currently a component type, then the type is replaced with a
+     * namespaced version of the component type, and the component type is set
+     * to null.
+     */
+    protected static Ds3ResponseType toUpdatedDs3ResponseType(
+            final Ds3ResponseType responseType) {
+        if (isEmpty(responseType.getComponentType())) {
+            return responseType;
+        }
+
+        return new Ds3ResponseType(
+                responseType.getComponentType() + NAMESPACE_ARRAY_RESPONSE_TYPES,
+                null);
+    }
+
+    /**
+     * Creates new types for generating encapsulating models. This occurs when a response
+     * handler has a return type with a component type. This is done to ensure that all
+     * response types can be properly parsed within the generated code.
+     * @param requests A Ds3Request
+     * @param types A list of Ds3Types
+     * @return The list of Ds3Types with the additionally generated types to handle the
+     * response payloads that return arrays
+     */
+    protected static ImmutableMap<String, Ds3Type> createEncapsulatingModelResponseTypes(
+            final ImmutableList<Ds3Request> requests,
+            final ImmutableMap<String, Ds3Type> types) {
+        if (isEmpty(requests)) {
+            LOG.info("No requests to generate encapsulating models for");
+            return types;
+        }
+        final ImmutableSet<String> componentTypes = getResponseComponentTypesFromRequests(requests);
+        if (isEmpty(componentTypes)) {
+            LOG.info("Requests do not contain any responses with component types");
+            return types;
+        }
+        final ImmutableMap.Builder<String, Ds3Type> builder = ImmutableMap.builder();
+        if (hasContent(types)) {
+            builder.putAll(types);
+        }
+        for (final String componentType : componentTypes) {
+            final Ds3Type ds3Type = toDs3Type(componentType);
+            builder.put(ds3Type.getName(), ds3Type);
+        }
+        return builder.build();
+    }
+
+    /**
+     * Converts a response Component Type into a simple Ds3Type that contains a
+     * list of the given component type, and where the new Ds3Type is namespaced.
+     */
+    protected static Ds3Type toDs3Type(final String componentType) {
+        if (isEmpty(componentType)) {
+            throw new IllegalArgumentException("Cannot generate Ds3Type from empty response component type");
+        }
+        return new Ds3Type(
+                componentType + NAMESPACE_ARRAY_RESPONSE_TYPES,
+                ImmutableList.of(
+                        new Ds3Element(
+                                stripPath(componentType),
+                                "array",
+                                componentType,
+                                null)),
+                null);
```

Comment:
`Ds3Type` should have an optional constructor that odes not require this extra null param